dss-api.yml

```diff
@@ -660,6 +660,250 @@ paths:
                     enum: [unhandled_exception, Forbidden, OAuthResponseProblem, not_found]
                 required:
                   - code
+  /collections:
+    get:
+      summary: Retrieve a user\'s collections.
+      description: |
+        Return a list of collections associated with the current user.
+      operationId: dss.api.collections.find
+      security:
+        - googAuth:
+            - https://www.googleapis.com/auth/userinfo.email
+      parameters:
+        - name: replica
+          in: query
+          description: Replica to fetch from. Can be one of aws, gcp, or azure.
+          required: true
+          type: string
+          enum: [aws, gcp, azure]
+      responses:
+        200:
+          description: OK
+          schema:
+            type: object
+            properties:
+              collections:
+                type: array
```

Comment:
Suggested: Looks like this route returns all collections of a user. Typically, directory routes should return a digest or listing of the items underneath. Or at best a list of sparsely populated items. I don't think we want this route to return all collections along with each collection's items.

Reply:
Requested: Just realized that this route raises NIE. I think we should not declare/document unimplemented API routes so as to not confuse or frustrate consumers. My suggestion above hints at ways this route might change.

Reply:
Sounds good. The intent here is to return a list of the current user's collections, paginated in the standard manner. However, this requires either making the primary index (S3 key prefix) be (f(user_email), collection_id), or making a secondary index (like `/collections/by-user/f(user_email)/collection_id` where the object contents are empty) - where f() may need to be sha256 for privacy, and the secondary index would provide a lookup hint only, with consistency verified by HEAD or GET on each collection). I'm not ready to implement either of those just yet, so I've removed the method for now and filed #1269 to track the related work.